docs/developers/clusterproviders.md

```diff
@@ -0,0 +1,234 @@
+# Cluster Providers
+
+A *ClusterProvider* is one of the three provider types in the openMCP architecture (the other two being *PlatformService* and *ServiceProvider*). ClusterProviders are responsible for managing kubernetes clusters and access to them, based on our [cluster API](https://github.com/openmcp-project/openmcp-operator/tree/main/api/clusters/v1alpha1).
+
+This document aims to describe the tasks of a ClusterProvider and the contract that it needs to fulfill in order to work within the openMCP ecosystem.
+
+## Deploying a ClusterProvider
+
+ClusterProviders are usually deployed via the [provider deployment](./provider_deployment.md) mechanism and need to stick to the corresponding contract.
+
+## Implementing a ClusterProvider
+
+### Provider Configuration
+
+Most ClusterProviders will probably require some form of configuration. Since the provider deployment does not allow passing in configuration via an argument to the binary directly, they need to read the configuration from a k8s resource. Depending on the provider, it might even allow multiple configuration resources and/or reconcile them instead of just reading them statically.
+
+### Cluster Profiles
+
+Out of the configuration(s), the ClusterProvider has to generate `ClusterProfile` resources. They serve as some kind of service discovery and look like this:
+```yaml
+apiVersion: clusters.openmcp.cloud/v1alpha1
+kind: ClusterProfile
+metadata:
+  name: default.gardener.mcpd-gcp-large
+spec:
+  providerConfigRef:
+    name: mcpd-gcp-large
+  providerRef:
+    name: gardener
+  supportedVersions:
+  - version: 1.33.3
+  - deprecated: true
+    version: 1.33.2
+  - version: 1.32.7
+  - deprecated: true
+    version: 1.32.6
+  - deprecated: true
+    version: 1.32.5
+  - deprecated: true
+    version: 1.32.4
+  - deprecated: true
+    version: 1.32.3
+  - deprecated: true
+    version: 1.32.2
+```
+
+`spec.providerRef` is the name of the ClusterProvider that created this `ClusterProfile`. It should be filled with the value that the provider received via its `--provider-name` argument.
```

Comment:
Maybe add a link to the provider_deployment section for the `provider-name` argument?